### DISCUSSION TITLE
Rsync correct code

### DISCUSSION THREAD

**Discussion by LiliVig** [pyControl/code#115] [Q&A] [answered]
Hi, I am trying to sync my pyControl breakout board whit a camera using the Rsync class, but I have trouble writing the correct syntax. That is what I write: 

class Rsync(pin, event = 'rsync', mean_IPI=5000, pulse_dur=50)
    
sync_output = Rsync(pin = board.BNC_1, mean_IPI=5000)
  
 rsync is defined in the events and board is defined as the breakout board (after importing from devices). 
What would be the correct syntax of those two lines?

**Comment by ThomasAkam** [comment]
Hi @LiliVig 

You just need to put the second code line in either your task file or hardware definition:

```python
board = Breakout_1_2() # Instantiate the breakout board object
sync_output = Rsync(pin = board.BNC_1, mean_IPI=5000) # Instantiate the Rsync output on breakout board BNC 1.
```

Then make sure the `'rsync'` event is included in your tasks `events` list and the sync pulses should be generated automatically when the task is running.

**Reply by LiliVig** [reply]
Hi @ThomasAkam

Thank you very much for your answer! Unfortunately, that didn't work:(

Stupid question, but do I still need to define Rsync as a class as in the first line? When I don't do that I get that Rsync is not defined, when I do it I get a syntax error. How would the proper way of writing it be?

Thanks again in advance!

**Reply by ThomasAkam** [reply, marked as answer]
You don't need to define the Rsync class but you do need to import it, which is usually done with `from devices import *`.

This is code for a minimal example task that generates rsync pulses:

```python
from pyControl.utility import *
from devices import *

# Define hardware (normally done in seperate hardware definition file).

board = Breakout_1_2()  # Instantiate the breakout board object
sync_output = Rsync(pin=board.BNC_1, mean_IPI=5000)  # Instantiate Rsync output

# States and events.

states = ["state_1"]

events = ["rsync"]

initial_state = "state_1"

# State behaviour functions

def state_1(event):
    pass
```

**Comment by LiliVig** [comment]
Hmm, I have already imported that. I was also wondering if that could be
the issue, I also didn’t find any definition of Rsync in devices or
utilities. How do I fix this?

**Comment by ThomasAkam** [comment]
The Rsync aligner class is defined in [pyControl.hardware](https://github.com/pyControl/code/blob/master/source/pyControl/hardware.py) but imported into devices in the devices [__init__](https://github.com/pyControl/code/blob/master/devices/__init__.py) file, so as long as you have `from devices import *` at the top of your task file you should be able to use it.  

Can you please upload your full task code that is not working and a screenshot of the pycontrol gui window with the error when you upload the task.

**Comment by LiliVig** [comment]
Hi,

Just before sending you the screenshots, I found the mistake, I was
overwriting on some other code that imported only the board and the LED
driver, not all devices, now the problem is fixed😅

Sorry to bother you and thank you for your help, at least now I understand
better how the Rsync works!

Best wishes!
